### PR TITLE
fix: Detect CT framework when unconfigured project opened with `--component`

### LINKED ISSUE #22777: Framework detection doesn't occur when opening Cy with `cypress open --component`
### Current behavior

I have a project configured for e2e and I want to set CT up.

I ran:

```sh
cypress open --component --browser=chrome
```

and the framework is not detected.

**Workaround**: If you click "Back" on the Testing Type Selection page in the launchpad and click on the "Component Testing" testing type card, it correctly detects the framework as Vite + Vue 3.

### Desired behavior

Detect the correct framework.

### Test code to reproduce

This project, at this commit (has e2e, but not ct yet): https://github.com/vueschool/vuejs-forge-the-project/commit/4b10ffd7d74b391832b763a6b846c0c87a383137

In this project, E2E was already setup before I tried to open Cy CT 10.2.0 for the first time on this machine.

### Cypress Version

10.2.0

### Other

_No response_

### PR DESCRIPTION
**Pull request by mike-plummer**
- Closes #22777 

### Additional details
The onboarding wizard automatically detects your CT framework and bundler when advancing from the testing type selection screen, but if you open the launchpad directly into CT using the `--component` flag this logic was being bypassed.  This PR performs the necessary wizard initialization if a testing type is already selected when `open` mode is being set up.

### Steps to test
1. Create a new project without any CT config
    `npm create vite@latest`
    choose React, remaining options can be anything
    `npm i` in project dir to install dependencies.
2. Using current production, open the project: `npx cypress open --component --project <project dir>`
3. Observe that launchpad opens to the "Project setup" screen and no framework is auto-selected
4. Close Cypress, checkout this branch
5. Using this branch, reopen the project: `yarn dev --component --project <project dir>`
6. Observe that launchpad auto-populates the framework and bundler fields with React + Vite

### How has the user experience changed?
**Before**
Screenshot 2023-03-30 at 8 54 05 AM

**After**
Screenshot 2023-03-30 at 9 12 24 AM

### PR Tasks
- [x] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/data-context/src/data/ProjectLifecycleManager.ts [hunk 1]
```diff
@@ -752,9 +752,14 @@ export class ProjectLifecycleManager {
...
-        // E2E doesn't have a wizard, so if we have a testing type on load we just create/update their cypress.config.js.
-        await this.ctx.actions.wizard.scaffoldTestingType()
+      if (!this.isTestingTypeConfigured(testingType)) {
+        if (testingType === 'e2e' && !this.ctx.migration.needsCypressJsonMigration()) {
+          // E2E doesn't have a wizard, so if we have a testing type on load we just create/update their cypress.config.js.
+          await this.ctx.actions.wizard.scaffoldTestingType()
+        } else if (testingType === 'component') {
+          await this.ctx.actions.wizard.detectFrameworks()
+          await this.ctx.actions.wizard.initialize()
+        }
```
Comment (warrensplayer) [inline packages/data-context/src/data/ProjectLifecycleManager.ts]: The guts of this function is now almost identical to the `setAndLoadCurrentTestingType` mutation.  Can these two areas be refactored to use the same code?  If it does not seem worth doing in this fix, then put some comments in both places to indicate that they should be updated together in the future.

**Review by warrensplayer — DISMISSED** [review]

File: packages/data-context/src/data/ProjectLifecycleManager.ts (hunk 1 above)
Comment (lmiller1990) [reply packages/data-context/src/data/ProjectLifecycleManager.ts]: If this is the case, I think we should just do the refactor right here right now -- realistically, are we going to slate time to revisit a low value TODO in the future?

**Review by lmiller1990 — APPROVED** [review]
Code looks good, won't block on merging, but if there's a quick refactor to be had, it sure would be nice to do it 💯

**Comment by marktnoonan** [comment]
This doesn't seem to be working for me with the project I tried, I'll test with a fresh react project like in the description, but let me know if my expectations are off in [this loom](https://www.loom.com/share/d4562ca5b0ab4ab5bf4c893f7f5cdc20)?

Update: I might have hit a strange cache issue where `yarn dev` does not load up to date code sometimes, because [it's working now](https://www.loom.com/share/098314a58bc34e44b35d3ea2418f52c1)...

**Comment by marktnoonan** [comment]
It works great when following the test steps in the issue description, I'm gonna try a fresh vue3-vite-ts project.

Screenshot 2023-03-31 at 1 53 28 PM

Fresh project was fine too...

Screenshot 2023-03-31 at 1 56 36 PM

Interesting that in the loom we do detect things by going out and back in 🤔

**Comment by mike-plummer** [comment]
@marktnoonan Very interesting. That happens every time with that specific project? Anything interesting pop in debug logs? I did notice one situation where the detection wouldn't work if the config file was invalid or otherwise couldn't be resolved correctly, but that typically caused other problems that were more noticeable.

**Comment by marktnoonan** [comment]
@mike-plummer I'll follow up in Slack, I updated [this comment](https://github.com/cypress-io/cypress/pull/26306#issuecomment-1492372845) with a link to the same exact project working correctly, I think maybe it's an issue with `yarn dev`.

**Review by marktnoonan — APPROVED** [review]

**Comment by marktnoonan** [comment]
I had already approved when I noticed I could actually consistently get this to not work on one project: [loom](https://www.loom.com/share/709591e07c1c4f64a659b032fd019ef4)

Commands:

`yarn gulp open --dev --component --project ../bug-reproductions/vite-detect/vue3-vite-ts` -- this is a fresh vite project just like the "steps to test" say but for Vue. This always works.

`yarn gulp open --dev --component --project ../cypress-component-testing-apps/vue3-vite-ts` -- this always fails, but can detect when you hit "back" and open CT again. This is using https://github.com/cypress-io/cypress-component-testing-apps/tree/main/vue3-vite-ts with CT config commented out.

Will see if I can capture useful logs.

**Comment by marktnoonan** [comment]
In the bad state the detection is actually fine, it's the UI not updating. If I refresh the launchpad I see the frameworks. So it's something where the gql reactivity is not working as expected?


Before refresh:
Screenshot 2023-03-31 at 4 03 11 PM

After refresh:
Screenshot 2023-03-31 at 4 04 10 PM

**Comment by mike-plummer** [comment]
The race condition Mark is seeing above has been identified but is a pre-existing issue that could have broader impacts to fix. Have written that up in #26388

**Pull request merged**